### ISSUE TITLE
Fix regression in 9.2.1 where stylelintrc.js is no longer supported

### ISSUE THREAD

**Issue by hudochenkov** [stylelint/stylelint#3315]
**Describe the bug? (A clear and concise description of what the bug is)**

`.stylelintrc.js` is not recognized anymore. This is new behaviour of cosmiconfig. [`searchPlaces`](https://github.com/davidtheclark/cosmiconfig/blob/fa5f52ef43e9eea14105fc13c6c6f49556c99c77/src/index.js#L25-L32) doesn't have `*rc.js` configs by default. I've opened [issue](https://github.com/davidtheclark/cosmiconfig/issues/144) in cosmiconfig repository.

**Which rule, if any, is this issue related to?**

Any.

**What CSS is needed to reproduce this issue?**

Any.

**What stylelint configuration is needed to reproduce this issue?**

`.stylelintrc.js`:

```js
module.exports = {
    rules: {
        indentation: 4,
    },
};

```

**Which version of stylelint are you using?**

`9.2.1`

**How are you running stylelint: CLI, PostCSS plugin, Node API?**

`stylelint "**/*.scss"`

**Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?**

No.

**What did you expect to happen?**

File to be linted.

**What actually happened (e.g. what warnings or errors you are getting)?**

The following warnings were flagged:

```shell
Error: No configuration provided for /Users/aleks/projects/stylelint-9-2-1-regression/test.scss
    at module.exports (/Users/aleks/projects/stylelint-9-2-1-regression/node_modules/stylelint/lib/utils/configurationError.js:8:28)
    at searchForConfig.then.then.config (/Users/aleks/projects/stylelint-9-2-1-regression/node_modules/stylelint/lib/getConfigForFile.js:49:15)
    at <anonymous>
```

**Comment by hudochenkov** [comment]
Fixed in cosmiconfig 5.0.4. Which is automatically fixed for our users.
